lib/galaxy/webapps/galaxy/api/tools.py

```diff
@@ -474,7 +475,17 @@ def _create(self, trans, payload, **kwd):
 
         # TODO: check for errors and ensure that output dataset(s) are available.
         output_datasets = vars.get('out_data', [])
-        rval = {'outputs': [], 'output_collections': [], 'jobs': [], 'implicit_collections': []}
+        rval = {'inputs': [], 'outputs': [], 'output_collections': [], 'jobs': [], 'implicit_collections': []}
+
+        def set_inputs(value, **kwargs):
```

Comment:
This seems a bit silly to me at first glance, the caller should know what the inputs are right? I don’t mind inserting this in here on the backend if it is easier for the GUI - but can we hide it behind some query parameter flag that makes it clear it isn’t part of the stable tool API?